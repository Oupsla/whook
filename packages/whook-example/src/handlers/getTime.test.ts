import initGetTime from './getTime';
import { TimeService } from 'common-services';

describe('getTime', () => {
  const time = jest.fn() as TimeService & jest.Mock<number>;

  it('should work', async () => {
    time.mockReturnValue(Date.parse('2014-01-26T00:00:00Z'));

    const getTime = await initGetTime({
      time,
    });
    const response = await getTime();

    expect({
      timeCalls: time.mock.calls,
      response,
    }).toMatchSnapshot();
  });
});
