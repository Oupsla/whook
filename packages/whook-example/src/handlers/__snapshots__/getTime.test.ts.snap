// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTime should work 1`] = `
Object {
  "response": Object {
    "body": Object {
      "time": 1390694400000,
    },
    "status": 200,
  },
  "timeCalls": Array [
    Array [],
  ],
}
`;
